konnectivity-client/pkg/client/client.go

```diff
@@ -340,6 +342,21 @@ func (t *grpcTunnel) Done() <-chan struct{} {
 	return t.done
 }
 
+// Send a best-effort DIAL_CLS request for the given dial ID.
+func (t *grpcTunnel) closeDial(dialID int64) {
```

Comment:
sendCloseDial?

Reply:
Also make sure we do not multi send based on our impl of net.Conn.Close()

Reply:
> Also make sure we do not multi send based on our impl of net.Conn.Close()

We only create a `conn{}` if the dial is successful, and this function is only ever called if the dial fails, so this should never happen. That said, if we did send 2 `DIAL_CLS` packets, the second one should be a no-op on the server.